### ISSUE TITLE
Checking consumptions of all requests breaks code

### ISSUE THREAD

**Issue by MStoykov** [go-openapi/runtime#119]
https://github.com/go-openapi/runtime/blob/1bd90842508346d9aae01eef6f2a062dcc8555e8/client/runtime.go#L341

The code above is breaking code that has being working for probably close to two years and a half. 
We have an internal api that has only consumes: "application/x-www-form-urlencoded". After upgrading runtime from 4812642f0b54441f8762a06e5c6c318a28917683 to current master this new check prevents a GET request to finish giving:
```
none of producers: map[application/json:0x7ddce0 application/xml:0x7deac0 text/plain:0x7de360 text/html:0x7de360 application/octet-stream:0x7dd470] registered. try application/x-www-form-urlencoded  
```
Deleting just this lines fixes the problem at least as far as my tests pass

Am I doing something wrong is the code generation doing something wrong or is the check too strict ?

**Comment by gregmarr** [comment]
Check out https://github.com/go-openapi/runtime/pull/49 and see if `type: formData` is what you should be using.

**Comment by MStoykov** [comment]
Adding the special case fixes it as well. 

I've made #120 with the change

**Comment by fredbi** [comment]
Closed by #120
